feat(marketing-analytics): impriving the ui for ma settings

jabahamondes opened the pull request:
## Problem

The production UI got out of control because of the big amount of sources and the UI in general was pretty basic.

## Changes

- Introduced  for adding new data sources.
- Added  for configuring column mappings for external tables.
- Created  for displaying lists with truncation support.
- Implemented  for managing pagination of displayed items.
- Added  component for visual feedback on source configuration status.
- Refactored  to utilize new components and improve UI/UX.
- Introduced  hook for managing sorted and paginated lists.

## Did you write or update any docs for this change?

- [ ] I've [added or updated the docs](https://posthog.com/handbook/engineering/writing-docs)
- [ ] I've reached out for help from the docs team
- [x] No docs needed for this change

## How did you test this code?

Manually + automatic snapshots

This is the result
<img width="954" alt="Screenshot 2025-06-12 at 12 14 14 PM" src="https://github.com/user-attachments/assets/63fe76a7-9a96-4cfb-9084-ddffad953fc3" />
<img width="948" alt="Screenshot 2025-06-12 at 12 14 24 PM" src="https://github.com/user-attachments/assets/f19f6c1e-1d3d-40b3-9157-1de0d8c22628" />
<img width="1053" alt="Screenshot 2025-06-12 at 12 23 24 PM" src="https://github.com/user-attachments/assets/18fda0e8-b23d-42d1-b32e-2e36bd41f4f6" />
<img width="948" alt="Screenshot 2025-06-12 at 12 14 24 PM" src="https://github.com/user-attachments/assets/acb5541b-3a26-4fac-9e5a-96cd7c155bae" />

Commit:
feat(marketing-analytics): add new components for source management and display

- Introduced  for adding new data sources.
- Added  for configuring column mappings for external tables.
- Created  for displaying lists with truncation support.
- Implemented  for managing pagination of displayed items.
- Added  component for visual feedback on source configuration status.
- Refactored  to utilize new components and improve UI/UX.
- Introduced  hook for managing sorted and paginated lists.

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

jabahamondes commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/settings/StatusIcon.tsx:
```diff
@@ -0,0 +1,18 @@
+        <div className="flex justify-center">
+            {status === 'success' && <IconCheck className="text-success text-lg" />}
+            {status === 'warning' && <IconWarning className="text-warning text-lg" />}
+            {status === 'error' && <IconX className="text-muted text-lg" />}
```
I want it to keep like that

jabahamondes commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/settings/AddSourceDropdown.tsx:
```diff
@@ -0,0 +1,40 @@
+                            <div className="flex items-center gap-2">
+                                <DataWarehouseSourceIcon type={source} />
+                                {source}
+                                <IconPlus className="text-muted" />
```
Didn't liked it

jabahamondes commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/settings/NativeExternalDataSourceConfiguration.tsx:
```diff
@@ -9,70 +9,142 @@ import { urls } from 'scenes/urls'
-    'ad_group_criterion',
-    'customer',
-]
+const NEEDED_FIELDS_FOR_MARKETING_ANALYTICS: Record<'GoogleAds' | 'MetaAds', string[]> = {
+    GoogleAds: ['ad', 'campaign', 'ad_group_ad', 'ad_group_criterion', 'customer'],
+    MetaAds: [],
+}
```
Intended bc it's not already supported

Commit:
fix: prs comments

Commit:
Merge remote-tracking branch 'origin/master' into feat/ui-improvement-ma-settings

Commit:
refactor: git pull

lricoy commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/settings/PaginationControls.tsx:
```diff
@@ -0,0 +1,41 @@
+    showAll: boolean
+    onToggleShowAll: () => void
+    totalCount: number
+    itemName: string // 'tables' or 'sources'
```
If those two are the only one possible: `'tables' | 'sources'`

jabahamondes replied:
Done ✅  🫶

lricoy commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/settings/AddSourceDropdown.tsx:
```diff
@@ -0,0 +1,40 @@
+
+interface AddSourceDropdownProps {
+    sources: ExternalDataSource['source_type'][] | ManualLinkSourceType[]
+    onSourceAdd: (source: any) => void // any because different files have different handler types
```
I would still suggest having a parent type /interface here rather than just defaulting to a `any`. It could even help with the other parts where you have `required_fields_for_marketing_analytics`.

No need to do this on this PR, though. You can refactor this when you have some more sources to define the base interface.

lricoy commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/settings/ColumnMappingModal.tsx:
```diff
@@ -0,0 +1,181 @@
+    }
+
+    // Get the current mapping from the sources_map to ensure we have the latest data
+    const currentSourceMap = sources_map[table.source_map_id] || {}
```
Is it ok to just default to `{}`?

jabahamondes replied:
Yes, it's fine, because we want to then check some fields on it

lricoy commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/settings/ColumnMappingModal.tsx:
```diff
@@ -0,0 +1,181 @@
+        )
+    }
+
+    const getFieldStatus = (fieldName: string): 'complete' | 'empty' => {
```
Moving those to a enum or type would be nice

jabahamondes replied:
Done ✅  🫶

lricoy commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/settings/ColumnMappingModal.tsx:
```diff
@@ -0,0 +1,181 @@
+                                </div>
+                                <div className="w-40">
+                                    <span className="font-medium">
+                                        {fieldName.replace(/_/g, ' ').replace(/\b\w/g, (l) => l.toUpperCase())}
```
Why the replacements?

jabahamondes replied:
So I use the column name as UI text. It would be like campaign_name into Campaign Name

lricoy commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/settings/NativeExternalDataSourceConfiguration.tsx:
```diff
@@ -9,70 +9,142 @@ import { urls } from 'scenes/urls'
     const marketingSources =
         dataWarehouseSources?.results.filter((source) => VALID_MARKETING_SOURCES.includes(source.source_type)) ?? []
 
+    // Helper functions to reduce duplication
```
Nice utilities

lricoy reviewed (approved):
This is shaping up quite nicely! Just some minor comments

Commit:
Merge remote-tracking branch 'origin/master' into feat/ui-improvement-ma-settings

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
refactor: Enhance type safety in marketing analytics components

- Updated  to use generics for better type handling of sources.
- Introduced  enum in  for clearer status management.
- Adjusted  to use  enum for item naming consistency.
- Refactored  and related components to leverage generics for source types, improving type safety across the board.

Commit:
Merge branch 'feat/ui-improvement-ma-settings' of github.com:PostHog/posthog into feat/ui-improvement-ma-settings
